hazelcast/test/src/HazelcastTests7.cpp

```diff
@@ -738,6 +738,18 @@ TEST_F(ClientQueueTest, testDrain)
     ASSERT_EQ("item1", list2[0]);
     ASSERT_EQ("item2", list2[1]);
     ASSERT_EQ("item3", list2[2]);
+
+    offer(2);
+    // drain to a non-empty vector
+    result = q->drain_to(list2).get();
+    ASSERT_EQ(2U, result);
+    // existing items
+    ASSERT_EQ("item1", list2[0]);
```

Comment:
let's double check the expected behaviour at Java.

Reply:
Also note that, this is just testing the existing behavior. I added the test just because I felt that it is missing. This test already passes on the current master